app/controllers/dashboard_controller.rb

```diff
@@ -14,6 +14,29 @@ def index
     redirect_to :action => 'show'
   end
 
+  # Redirect to cockpit with an api auth token
+  def cockpit_redirect
+    # Behind apache use paths only
+    if request.env['HTTP_X_FORWARDED_FOR']
+      url = URI("/cws/")
+    # No apache, use current host unless external url configured
+    else
+      host = URI.parse(request.original_url).host
+      conf = MiqCockpitWsWorker.worker_settings()
+      url = MiqCockpit::WS.url(conf || {}, host)
```

Comment:
Be careful here with ipv6 addresses. URI#host will not unwrap `[::1]` for use with socket connections.

If you're using it for building a uri, always use URI::HTTP.build or URI::Generic.build with `:host => "::1".  If you need the unwrapped `::1`

```
irb(main):007:0* URI::HTTP.build(:host => '::1').to_s
=> "http://[::1]"
irb(main):008:0> URI::HTTP.build(:host => '::1').host
=> "[::1]"
irb(main):009:0> URI::HTTP.build(:host => '::1').hostname # for socket connections
=> "::1"
irb(main):010:0> URI::HTTP.build(:host => '127.0.0.1').to_s
=> "http://127.0.0.1"
irb(main):011:0> URI::HTTP.build(:host => '127.0.0.1').host
=> "127.0.0.1"
irb(main):012:0> URI::HTTP.build(:host => '127.0.0.1').hostname
=> "127.0.0.1"
```